src/messages/MOSDPGInfo.h

```diff
@@ -58,98 +58,11 @@ class MOSDPGInfo : public Message {
   }
 
   void encode_payload(uint64_t features) override {
-    if (HAVE_FEATURE(features, SERVER_LUMINOUS)) {
-      header.version = HEAD_VERSION;
```

Comment:
IIUC the assignment is only needed if we have a path that sets it to a lower value also, right?  So it would get added back if/when we have a new compat encoding.  (Or maybe that's too fragile and we should assign it unconditionally defensively.. in which case we should probably make the HEAD_VERSION accessible generically and have it reset by common code instead of every Message implementation?)

Reply:
oh, right. since we won't set `header.version` according to `feature` anymore. so we are safe here.